### PR TITLE
add release procedure notes, remove old steps

### PR DESCRIPTION
**Pull request by balos1**

### CODE REVIEW

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 1]
```diff
@@ -65,50 +65,69 @@ This is a list of tasks that need to be done before a SUNDIALS release.
 The order is bottom-up starting with solver source files and ending with
 web pages.
 
-#. Create release branch from trunk in the Git repository
+#. Create release branch ``release/vX.Y.Z`` from develop in the git repo.
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:68]:
```suggestion
#. Create release branch ``release/vX.Y.Z`` from ``develop`` in the git repo.
```

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 2]
```diff
@@ -65,50 +65,69 @@ This is a list of tasks that need to be done before a SUNDIALS release.
...
+
+#. Update version numbers and release date information using the ``updateVerson.sh``
+   script.
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:71]: This has to be run from the `scripts` directory (for now, I have a branch that will remove this requirement)
```suggestion
   script.
   
   .. code-block:: shell
      pushd scripts/ && ./updateVersion.sh && popd
```

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 3]
```diff
@@ -65,50 +65,69 @@ This is a list of tasks that need to be done before a SUNDIALS release.
...
-   The following files are no longer maintained:
-
-   * ``html/main.html`` (This is no longer maintained as of at least 2016)
-   * ``sundialsTB/install_STB.m`` (This is no longer maintained as of 2016)
+Release Procedure
+=================
 
-#. Update version numbers of third party libraries in the Install Guide
-   in doc directory.
+#. Once the release PR is passing all tests and has been approved, merge it.
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:86]:
```suggestion
#. Once the release PR is passing all tests and has been approved, merge it. Like all
   merges to ``develop``, this should be done with the "Squash and merge" option.
```

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 4]
```diff
@@ -65,50 +65,69 @@ This is a list of tasks that need to be done before a SUNDIALS release.
...
    that they are deprecated).
 
-#. Regenerate the Fortran 2003 interfaces. It is possible nothing will be updated.
-   This is done by running ``make all32 all64`` in the ``swig/`` directory.
+#. Update version numbers of third party libraries in the Install Guide
+   in doc directory.
 
-#. Update the "Changes in ..." sections in all user guides. The changes should be
-   sorted so that major new features are above bug fixes.
+#. Open a pull request from the release branch to ``develop``.
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:81]:
```suggestion
#. Open a pull request from the release branch to ``develop`` with the title
   "Release: vX.Y.Z" and description "SUNDIALS Release vX.Y.Z".
```

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 5]
```diff
@@ -65,50 +65,69 @@ This is a list of tasks that need to be done before a SUNDIALS release.
...
+   a GitHub PR (so that the merge is a fast-forward). The steps are as follows:
+
+   .. code-block:: shell
+
+      git checkout develop
+      git pull # develop should be up to date with origin/develop now
+      git checkout main
+      git pull # main should be up to date with origin/main now
+      git merge --ff-only develop # we want to do a fast-forward merge (no merge commit)
+      git tag -m 'SUNDIALS Release vX.Y.Z' vX.Y.Z
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:98]: I've been using annotated tags for releases
```suggestion
      git tag -a vX.Y.Z -m 'SUNDIALS Release vX.Y.Z' vX.Y.Z
```

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 6]
```diff
@@ -65,50 +65,69 @@ This is a list of tasks that need to be done before a SUNDIALS release.
...
+      git push --tags origin main
 
-#. Create the new distribution tarballs using the ``tarscript`` shell script
-   under the ``scripts`` directory. This also compiles the documents (user
-   guides and example docs) and creates all tarballs in their final form,
-   appropriate for uploading to the website.
+   .. note::
+
+      The final step (pushing to main) requires changing the GitHub repository settings to
+      allow whoever is doing the push an exception in the ``main`` branch protection rules.
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:104]:
```suggestion
      The final step (pushing to main) requires temporarily changing the GitHub
      repository settings to allow whoever is doing the push an exception in the
      ``main`` branch protection rules.
      
   .. danger::
   
      Remember to remove this exception to the2 branch protection rules after making
      the push to update ``main``.
```

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 7]
```diff
@@ -65,50 +65,69 @@ This is a list of tasks that need to be done before a SUNDIALS release.
...
+#. Once readthedocs finishes building the new release, create the tarballs *on a Linux machine*.
+   Use the ``tarscript`` shell script under the ``scripts`` directory. This also compiles the documents
+   (user guides and example docs) and creates all tarballs in their final form, appropriate for uploading
+   as artifacts to the GitHub release.
+
+   .. warning::
+
+	   Creating the tarballs on a Mac can cause issues. Furthermore, it is important to wait
+      to create the tarballs until readthedocs finishes building the new release docs so
+	   that cross-references have valid links.
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:115]: Looks like a tab v spaces issue

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 8]
```diff
@@ -65,50 +65,69 @@ This is a list of tasks that need to be done before a SUNDIALS release.
...
+
+#. Draft the release on GitHub and attach the tarballs as well as the example documentation PDFs.
+   The description of the release is just a copy of the ``CHANGELOG.md`` notes for the release with
+   hard line-wraps removed.
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:119]:
```suggestion
#. Draft the release on GitHub with the title "SUNDIALS vX.Y.Z" and attach the tarballs
   as well as the example documentation PDFs. The description of the release is just a
   copy of the ``CHANGELOG.md`` notes for the release with hard line-wraps removed.
```

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 9]
```diff
@@ -118,22 +137,13 @@ web pages.
...
-          https://computing-staging.llnl.gov/projects/sundials/release-history
-        * Add list of primary changes and child page containing
-          full list of changes (this information is shared with
-          "Changes in ..." sections of user guides).
-
       * Edit Download Page:
         https://computing-staging.llnl.gov/projects/sundials/sundials-software
 
         * Update main download table with links to new versions of solvers
         * The documentation links do not need to be updated (they point to the PDFs on the GitHub main branch)
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:144]: I'm guessing the link for the example documentation PDFs needs to be updated now that we upload them with the tarballs.

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 10]
```diff
@@ -146,8 +156,6 @@ web pages.
 #. After final push, ensure web content and behavior is as expected on the main
    page: http://computing.llnl.gov/projects/sundials
 
-#. Tag the release
-
 
 **Old steps for maintaianed code:**
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:160]: Should we remove the steps below?

**Review by gardner48 — CHANGES_REQUESTED** [review]

File: doc/superbuild/source/developers/releases/Checklist.rst (hunk 10 above)
Comment (balos1) [reply doc/superbuild/source/developers/releases/Checklist.rst:171]: Yeah, I think we can do that now.

File: doc/superbuild/source/developers/releases/Checklist.rst [hunk 11]
```diff
@@ -95,13 +100,19 @@ Release Procedure
       git checkout main
       git pull # main should be up to date with origin/main now
       git merge --ff-only develop # we want to do a fast-forward merge (no merge commit)
-      git tag -m 'SUNDIALS Release vX.Y.Z' vX.Y.Z
+      git tag -a vX.Y.Z -m 'SUNDIALS Release vX.Y.Z' vX.Y.Z
```
Comment (gardner48) [inline doc/superbuild/source/developers/releases/Checklist.rst:103]: Forgot to remove the trailing `vX.Y.Z` in my suggestion
```suggestion
      git tag -a vX.Y.Z -m 'SUNDIALS Release vX.Y.Z'
```

**Review by gardner48 — CHANGES_REQUESTED** [review]
One minor fix (from my own suggestion)

**Review by gardner48 — APPROVED** [review]

**Pull request merged**